pkg/executables/kind.go

```diff
@@ -181,20 +181,21 @@ func (k *Kind) DeleteBootstrapCluster(ctx context.Context, cluster *types.Cluste
 
 func (k *Kind) setupExecConfig(clusterSpec *cluster.Spec) error {
 	bundle := clusterSpec.VersionsBundle
+	registry := clusterSpec.Cluster.RegistryMirror()
 	k.execConfig = &kindExecConfig{
-		KindImage:            urls.ReplaceHost(bundle.EksD.KindNode.VersionedImage(), clusterSpec.Cluster.RegistryMirror()),
-		KubernetesRepository: bundle.KubeDistro.Kubernetes.Repository,
+		KindImage:            urls.ReplaceHost(bundle.EksD.KindNode.VersionedImage(), registry),
+		KubernetesRepository: urls.ReplaceHost(bundle.KubeDistro.Kubernetes.Repository, registry),
```

Comment:
this another place that can benefit from a `ImageRegistryRepository` (whatever other name) that takes an image url and return another one with any necessary replacement done based on the registry mirror mapping configs